code/modules/mob/living/carbon/xenomorph/xeno_defines.dm

```diff
@@ -907,7 +909,19 @@
 		for(var/obj/item/alien_embryo/embryo in potential_host)
 			embryo.hivenumber = XENO_HIVE_FORSAKEN
 		potential_host.update_med_icon()
+	var/shipside_humans_count = 0
+	var/xenos_count = 0
+	for(var/mob/living/carbon/human/current_human as anything in GLOB.alive_human_list)
+		if(isspecieshuman(current_human) || isspeciessynth(current_human))
+			var/turf/turf = get_turf(current_human)
+			if(is_mainship_level(turf?.z))
+				shipside_humans_count++
+	for(var/mob/living/carbon/xenomorph/xenomorph as anything in totalXenos)
+		if(isfacehugger(xenomorph))
+			continue
+		xenos_count++
 	hijack_burrowed_surge = TRUE
+	hijack_burrowed_left = max(n_ceil(shipside_humans_count * 0.5) - xenos_count, 5)
```

Comment:
To compensate for the difference between a combat marine and a non-combat marine consider weighing the roles like is done with `latejoin_tally` via `RoleAuthority.calculate_role_weight`

Reply:
Done, I think? Should work this way